src/path/sentry_path_windows.c

```diff
@@ -30,20 +30,18 @@ struct sentry_pathiter_s {
 static size_t
 write_loop(FILE *f, const char *buf, size_t buf_len)
 {
-    size_t offset = 0;
-    size_t remaining = buf_len;
-    while (true) {
-        if (remaining == 0) {
-            break;
-        }
-        size_t n = fwrite(buf + offset, 1, remaining, f);
-        if (n == 0) {
+    while (buf_len > 0) {
+        size_t n = fwrite(buf, 1, buf_len, f);
+        if (n == 0 && errno == EINVAL) {
+            continue;
+        } else if (n < buf_len) {
             break;
         }
-        offset += n;
-        remaining -= n;
+        buf += n;
+        buf_len -= n;
     }
-    return remaining;
+    fflush(f);
```

Comment:
Is calling `fflush` after an error is still valid or do we need to do that inside of the loop?

Reply:
Docs don’t mention anything about it. I also think the flush should not be in a loop, since its potentially slow.